### PR TITLE
Add logging to try and find a possible connection race condition

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
For #5388. I looked into it but nothing is sticking out as being a possible reason why this could be happening. So adding some logging so that hopefully if someone hits this we'll have something to go off of.

### CODE REVIEW

File: src/sql/platform/connection/common/connectionStatusManager.ts [hunk 1]
```diff
@@ -109,6 +122,15 @@ export class ConnectionStatusManager {
 
 	public onConnectionComplete(summary: azdata.ConnectionInfoSummary): ConnectionManagementInfo {
 		let connection = this._connections[summary.ownerUri];
+		if (!connection) {
+			this._logService.info(`########## OnConnectionComplete but no connection found '${summary.ownerUri}' Connections = [${Object.keys(this._connections)}]`);
+			this._notificationService.notify({
+				severity: Severity.Error,
+				message: `An unexpected error occurred while connecting. Please [file an issue](command:workbench.action.openIssueReporter) with the title 'Unexpected Error Occurred while Connecting' and include the log file [${join(this._environmentService.logsPath, 'renderer1.log')}](command:workbench.action.openLogsFolder)`
```
Comment (Charles-Gagnon) [inline src/sql/platform/connection/common/connectionStatusManager.ts]: Plus I got to try the new command-link and it works great!

**Comment by Charles-Gagnon** [comment]
@kburtram Do you have any thoughts on this? Worth adding this in or just keep trying to get a local repro?

File: src/sql/platform/connection/common/connectionStatusManager.ts [hunk 2]
```diff
@@ -7,15 +7,23 @@ import { ConnectionManagementInfo } from 'sql/platform/connection/common/connect
 import { ICapabilitiesService } from 'sql/platform/capabilities/common/capabilitiesService';
 import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
 import { IConnectionProfile } from 'sql/platform/connection/common/interfaces';
+import { StopWatch } from 'vs/base/common/stopwatch';
```
Comment (ranasaria) [inline src/sql/platform/connection/common/connectionStatusManager.ts]: curious: why move this?

Comment (Charles-Gagnon) [reply src/sql/platform/connection/common/connectionStatusManager.ts]: I don't recall doing this on purpose, could have been VS Code moving stuff around with the auto-import

File: src/sql/platform/connection/common/connectionStatusManager.ts [hunk 3]
```diff
@@ -109,6 +122,15 @@ export class ConnectionStatusManager {
 
 	public onConnectionComplete(summary: azdata.ConnectionInfoSummary): ConnectionManagementInfo {
 		let connection = this._connections[summary.ownerUri];
+		if (!connection) {
+			this._logService.info(`########## OnConnectionComplete but no connection found '${summary.ownerUri}' Connections = [${Object.keys(this._connections)}]`);
```
Comment (kburtram) [inline src/sql/platform/connection/common/connectionStatusManager.ts]: I don't know if we're following the `##########` convention in the our comments anywhere else.  Is there a reason to do so here?

**Pull request merged**